src/components/ReportActionItem/TaskPreview.tsx

```diff
@@ -24,141 +21,136 @@ import * as LocalePhoneNumber from '@libs/LocalePhoneNumber';
 import ONYXKEYS from '@src/ONYXKEYS';
 import ROUTES from '@src/ROUTES';
+import type {Policy, Report, ReportAction} from '@src/types/onyx';
+import {isNotEmptyObject} from '@src/types/utils/EmptyObject';
 
-const propTypes = {
-    /** The ID of the associated taskReport */
-    taskReportID: PropTypes.string.isRequired,
-
-    /** Whether the task preview is hovered so we can modify its style */
-    isHovered: PropTypes.bool,
-
-    /** The linked reportAction */
-    action: PropTypes.shape(reportActionPropTypes).isRequired,
+type PolicyRole = {
+    /** The role of current user */
+    role: string;
+};
 
+type TaskPreviewOnyxProps = {
     /* Onyx Props */
 
-    taskReport: PropTypes.shape({
-        /** Title of the task */
-        reportName: PropTypes.string,
-
-        /** AccountID of the manager in this iou report */
-        managerID: PropTypes.number,
-
-        /** AccountID of the creator of this iou report */
-        ownerAccountID: PropTypes.number,
-    }),
+    /* current report of TaskPreview */
+    taskReport: OnyxEntry<Report>;
 
     /** The policy of root parent report */
-    rootParentReportpolicy: PropTypes.shape({
-        /** The role of current user */
-        role: PropTypes.string,
-    }),
-
-    /** The chat report associated with taskReport */
-    chatReportID: PropTypes.string.isRequired,
-
-    /** Popover context menu anchor, used for showing context menu */
-    contextMenuAnchor: refPropTypes,
-
-    /** Callback for updating context menu active state, used for showing context menu */
-    checkIfContextMenuActive: PropTypes.func,
-
-    /* Onyx Props */
-    ...withLocalizePropTypes,
-
-    ...withCurrentUserPersonalDetailsPropTypes,
-};
-
-const defaultProps = {
-    ...withCurrentUserPersonalDetailsDefaultProps,
-    taskReport: {},
-    rootParentReportpolicy: {},
-    isHovered: false,
+    rootParentReportpolicy: OnyxEntry<PolicyRole>;
 };
 
-function TaskPreview(props) {
+type TaskPreviewProps = WithCurrentUserPersonalDetailsProps &
+    TaskPreviewOnyxProps & {
+        /** The ID of the associated policy */
+        // eslint-disable-next-line react/no-unused-prop-types
+        policyID: string;
+        /** The ID of the associated taskReport */
+        taskReportID: string;
+
+        /** Whether the task preview is hovered so we can modify its style */
+        isHovered: boolean;
+
+        /** The linked reportAction */
+        action: OnyxEntry<ReportAction>;
+
+        /** The chat report associated with taskReport */
+        chatReportID: string;
+
+        /** Popover context menu anchor, used for showing context menu */
+        contextMenuAnchor: Element;
+
+        /** Callback for updating context menu active state, used for showing context menu */
+        checkIfContextMenuActive: () => void;
+    };
+
+function TaskPreview({
+    taskReport,
+    taskReportID,
+    action,
+    contextMenuAnchor,
+    chatReportID,
+    checkIfContextMenuActive,
+    isHovered,
```

Comment:
Missing default value here:
```suggestion
    isHovered = false,
```